src/test/regress/sql/multi_subquery_misc.sql

```diff
@@ -0,0 +1,155 @@
+-- multi subquery pushdown misc aims to test subquery pushdown queries with
+--   (i)   Prepared statements
+--   (ii)  PL/PGSQL functions
+--   (iii) SQL functions
+
+-- the tables that are used depends to multi_insert_select_behavioral_analytics_create_table.sql
+
+-- We don't need shard id sequence here, so commented out to prevent conflicts with concurrent tests
+SET citus.enable_router_execution TO false;
+
+PREPARE prepared_subquery_1 AS
+SELECT
+    user_id,
+    user_lastseen,
+    array_length(event_array, 1)
+FROM (
+    SELECT
+        user_id,
+        max(u.time) as user_lastseen,
+        array_agg(event_type ORDER BY u.time) AS event_array
+    FROM (
+        SELECT user_id, time
+        FROM users_table
+        WHERE
+        user_id >= 10 AND
+        user_id <= 70 AND
+        users_table.value_1 > 10 AND users_table.value_1 < 12
+        ) u LEFT JOIN LATERAL (
+          SELECT event_type, time
+          FROM events_table
+          WHERE user_id = u.user_id AND
+          events_table.event_type > 10 AND events_table.event_type < 12
+        ) t ON true
+        GROUP BY user_id
+) AS shard_union
+ORDER BY user_lastseen DESC, user_id;
+
+EXECUTE prepared_subquery_1;
+
+
+PREPARE prepared_subquery_2(int, int) AS
+SELECT
+    user_id,
+    user_lastseen,
+    array_length(event_array, 1)
+FROM (
+    SELECT
+        user_id,
+        max(u.time) as user_lastseen,
+        array_agg(event_type ORDER BY u.time) AS event_array
+    FROM (
+        SELECT user_id, time
+        FROM users_table
+        WHERE
+        user_id >= $1 AND
+        user_id <= $2 AND
+        users_table.value_1 > 10 AND users_table.value_1 < 12
+        ) u LEFT JOIN LATERAL (
+          SELECT event_type, time
+          FROM events_table
+          WHERE user_id = u.user_id AND
+          events_table.event_type > 10 AND events_table.event_type < 12
+        ) t ON true
+        GROUP BY user_id
+) AS shard_union
+ORDER BY user_lastseen DESC, user_id;
+
+-- should be fine with more than five executions
+EXECUTE prepared_subquery_2(10, 70);
+EXECUTE prepared_subquery_2(10, 70);
+EXECUTE prepared_subquery_2(10, 70);
+EXECUTE prepared_subquery_2(10, 70);
+EXECUTE prepared_subquery_2(10, 70);
+EXECUTE prepared_subquery_2(10, 70);
+EXECUTE prepared_subquery_2(10, 70);
+
+-- prepared statements with subqueries in WHERE clause
+PREPARE prepared_subquery_3(int, int, int, int, int, int) AS
+SELECT user_id
+FROM users_table
+WHERE user_id IN (SELECT user_id FROM users_table WHERE value_1 >= $4 AND value_1 <= $3)
+    AND user_id IN (SELECT user_id FROM users_table WHERE value_1 >= $5 AND value_1 <= $6)
+    AND user_id IN (SELECT user_id FROM users_table WHERE  value_1 >= $1 AND value_1 <= $2)
+GROUP BY 
+  user_id
+ORDER BY
+  user_id DESC
+  LIMIT 5;
+
+EXECUTE prepared_subquery_3(50, 60, 20, 10, 30, 40);
```

Comment:
I though we already have the tests with >5 times executed. This is mostly to ensure that it works with subqueries in WHERE clause as well. 
But, thinking more, I'll repeat this 6 times as well.